all: fix grammar, typos, reformat tables

J4bbi opened the pull request:
### Description

This PR fixes obvious grammatical mistakes, typos and reformats some tables to make the markdown more readabe/correct.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

fenekku reviewed (commented):
Thank you so much for this! This must have been a gradual investment over time and we really appreciate it. 

There is a couple of further adjustments I added. I will take another look after and then merge. 

This will really level up the professionalism of the project. Thanks so much again!

J4bbi commented on docs/reference/metadata.md:
```diff
@@ -83,15 +83,15 @@ A record stores information about two internal PIDs:
 
-| Field |   Description   |
-|:-----:|:----------------|
+|     Field     | Description                                 |
```
I've left the centering, just because it's more faff to revert it if that's okay

J4bbi commented:
@fenekku , thanks for the review, I've responded to all your comments, and I also changed a couple of other thing I noticed

I removed the last sentence 
"All of this can be further complicated if you run container images or virtual machines" from virtualenvs.md because I didn't think it added much value.

fenekku commented on docs/develop/architecture/software.md:
```diff
@@ -24,13 +24,13 @@ The diagram below shows a simplified view of the data flow in the architecture.
 ![Architecture layers](../img/architecture.svg)
 
-*The presentation layer* parses incoming requests and routes them to service layer. This involves sending and receiving data in multiple different formats and translating these into an internal representation, as well as e.g. parsing arguments from an HTTP request (e.g parsing the query string parameters).
+*The presentation layer* parses incoming requests and routes them to service layer. This involves sending and receiving data in multiple different formats and translating these into an internal representation, as well as e.g., parsing arguments from an HTTP request (e.g., parsing the query string parameters).
```
```suggestion
*The presentation layer* parses incoming requests and routes them to service layer. This involves sending and receiving data in multiple different formats and translating these into an internal representation. For instance, this includes parsing arguments from an HTTP request e.g., parsing the query string parameters.
```

fenekku commented on docs/develop/architecture/software.md:
```diff
@@ -24,13 +24,13 @@ The diagram below shows a simplified view of the data flow in the architecture.
+The data flow between the layers is strictly limited to some few well-defined objects to ensure a clean separation of concerns. The presentation layer communicates with the service layer via a record projection (i.e. a view of a record localised to a specific identity). The service layer communicates with the data access layer via a record entity that provides data abstraction, syntactic data val ...
 
 !!! tip "Tip: Where do you belong?"
 
```
Just below is another fix to do: "where you code belongs" -> "where your code belongs"

fenekku commented on docs/develop/getting-started/virtualenvs.md:
```diff
@@ -24,7 +24,7 @@ programs (i.e. a distribution provides the ``python`` command).
 ### Distribution version
 
 Each distribution comes in multiple versions that's able to interpret the
-Python programming language - e.g Python 3.8, 3.9 etc.
+Python programming language - e.g., Python 3.8 and 3.9
```
`Each distribution supports multiple version of the Python programming language e.g., Python 3.8 and Python 3.9`

(can't create a suggestion for unmodified line, but let's reword it too)

fenekku commented on docs/develop/getting-started/virtualenvs.md:
```diff
@@ -57,15 +57,15 @@ Thus in addition to having multiple ``python`` commands, you can have multiple
 
-This means that a virtual environment has it's own ``python`` and ``pip``
+This means that a virtual environment has its own ``python`` and ``pip``
 commands that are *enhanced aliases* to the distributions ``python`` and ``pip``
```
Don't know too much about that *enhanced aliases* business to be honest... I think cutting it at:

```
This means that a virtual environment has its own ``python`` and ``pip`` commands.
```

is good enough.

fenekku commented on docs/maintenance/release-management.md:
```diff
@@ -86,7 +86,7 @@ For modules in ``vX.Y.Z``, the new version is ``v(X+1).0.0.dev0``.
 - Cookiecutter-Invenio-RDM:
     - Merge everything to ``master``.
-    - Create new version branch from ``master`` using the pattern ``vX.Y`` (e.g if Invenio-App-RDM is v1.0.0, the branch should be named ``v1.0``).
+    - Create a new version branch from ``master`` using the pattern ``vX.Y`` (e.g. if Invenio-App-RDM is v1.0.0, then the branch should be named ``v1.0``).
```
```suggestion
    - Create a new version branch from ``master`` using the pattern ``vX.Y`` (e.g., if Invenio-App-RDM is v1.0.0, then the branch should be named ``v1.0``).
```

J4bbi commented:
@fenekku , further comments addressed

fenekku commented:
![image](https://github.com/inveniosoftware/docs-invenio-rdm/assets/1932651/dc47a00c-81ac-4ce5-b336-57757ebd0fd1)
